pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -59,6 +61,51 @@ func NewActuator(params ActuatorParams) *Actuator {
 	}
 }
 
+func (a *Actuator) getClusterMachines(ms *actuators.MachineScope, cluster *clusterv1.Cluster) (*clusterv1.MachineList, error) {
+	clusterMachines, err := ms.MachineClient.List(v1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get machinesList for cluster %q in namespace %q: %v", cluster.Name, cluster.Namespace, err)
+	}
+	return clusterMachines, nil
+}
+
+func (a *Actuator) getControlPlaneMachines(machineList *clusterv1.MachineList) []*clusterv1.Machine {
+	var cpm []*clusterv1.Machine
+	for _, m := range machineList.Items {
+		if m.Spec.Versions.ControlPlane != "" {
+			cpm = append(cpm, m.DeepCopy())
+		}
+	}
+	return cpm
+}
+
+func isMachinesSame(m1 *clusterv1.Machine, m2 *clusterv1.Machine) bool {
+	return m1.Name == m2.Name && m1.Namespace == m2.Namespace
+}
+
+func (a *Actuator) isNodeJoin(controlPlaneMachines []*clusterv1.Machine, newMachine *clusterv1.Machine) (join bool, err error) {
+	err = nil
+	switch newMachine.ObjectMeta.Labels["set"] {
+	case "node":
+		join = true
+	case "controlplane":
+		join = true
+		// join = false, if:
+		//		1. len(controlPlaneMachines) == 1 && controlPlaneMachines[0] == newMachine
+		if len(controlPlaneMachines) == 1 && isMachinesSame(controlPlaneMachines[0], newMachine) {
+			join = false
```

Comment:
directly return false and nil here. after this `if` block, we can directly return true and nil instead of setting default values and implicitly returning